discovery_engine/lib/src/domain/engine/mock_engine.dart

```diff
@@ -222,6 +224,13 @@ class MockEngine implements Engine {
     }
   }
 
+  @override
+  Future<List<TrendingTopic>> getTrendingTopics() async {
+    _incrementCount('getTrendingTopics');
+    // TODO: implement mock response
+    return [];
```

Comment:
:art: similarly to the engine implementation we could throw an error until it's implemented to avoid missing this.
```suggestion
    throw UnimplementedError('TODO: implement mock response');
```